src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -161,7 +162,59 @@ def register_submission(
                 get_drc=get_drc,
                 language=language_code_2b,
             )
-            record_data["csv_url"] = submission_csv_document["url"]
+            csv_url = submission_csv_document["url"]
+
+        base_render_context = {
+            "_submission": submission,
+            **get_variables_for_context(submission),
+            "productaanvraag_type": str(options["productaanvraag_type"]),
+            # nested for namespacing note: other templates and context expose all submission
+            # variables in the top level namespace, but that is due for refactor
+            "submission": {
+                "public_reference": submission.public_registration_reference,
+                "kenmerk": str(submission.uuid),
+                "language_code": submission.language_code,
+                "attachments": attachments,
+                "pdf_url": document["url"],
+                "csv_url": csv_url,
+            },
+        }
+
+        template = (
+            options.get("content_json") or ObjectsAPIConfig.get_solo().content_json
+        )
+
+        record_data = render_from_string(
+            template,
+            context={**base_render_context},
+            disable_autoescape=False,
+            backend=openforms_backend,
+        )
+
+        if sys.getsizeof(record_data) > settings.OBJECTS_API_DATA_SIZE_LIMIT:
+            logger.error(
+                "Content JSON field exceeded size of %s with total size: %s",
+                settings.OBJECTS_API_DATA_SIZE_LIMIT,
+                sys.getsizeof(record_data),
+            )
+            raise SuspiciousOperation(
+                _(
+                    "Content JSON field exceeded size of %(max)s with total size: %(total)s.",
+                )
+                % {
+                    "max": settings.OBJECTS_API_DATA_SIZE_LIMIT,
+                    "total": sys.getsizeof(record_data),
+                }
+            )
```

Comment:
* we don't translate exception messages, as they're not user facing.
* you can also make the file sizes a bit more human-readable using Django's `filesizeformat` filter
* finally, using the walrus operator (`:=`) avoids re-calculating the same thing

```suggestion
from django.template.defaultfilters import filesizeformat
        
        if (data_size := sys.getsizeof(record_data)) > settings.OBJECTS_API_DATA_SIZE_LIMIT:
            formatted_size = filesizeformat(data_size)
            max_size = filesizeformat(settings.OBJECTS_API_DATA_SIZE_LIMIT)
            raise SuspiciousOperation(
                f"Templated out content JSON exceeds the maximum size {max_size} ("
                f"it is {formatted_size})."
            )
```